specs/ExtendedAcceleratorKeyPressed.md

```diff
@@ -0,0 +1,128 @@
+                        }
+                        return S_OK;
+                    })
+                    .Get(),
+                &m_acceleratorKeyPressedToken));
+
+        }
+    }
+```
+
+## C#
+```c#
+    CoreWebView2Settings _webViewSettings = webView.CoreWebView2.Settings;
+    // All browser accelerator keys are disabled with this setting
+    _webViewSettings.AreBrowserAcceleratorKeysEnabled = false;
+
+    webView.CoreWebView2.AcceleratorKeyPressed += WebView2Controller_AcceleratorKeyPressed;
+
+    void WebView2Controller_AcceleratorKeyPressed(object sender, CoreWebView2AcceleratorKeyPressedEventArgs e)
+    {
+        switch (e.KeyEventKind)
+        {
+            case CoreWebView2KeyEventKind.KeyDown:
+            case CoreWebView2KeyEventKind.SystemKeyDown:
+                {
+                    Keys keyData = (Keys)(e.VirtualKey);
+                    // Allow the browser to process F7 key
+                    if (e.VirtualKey == Key.F7) 
+                    {
+                        e.AllowBrowserHandle = true;
+                    }
+                    break;
+                }
+            
+        }
+    }
+
+```
+
+# Remarks
+By default, `AreBrowserAcceleratorKeyEnabled` is `TRUE` and default value for `AllowBrowserHandle` is `TRUE`. When developers set `AreBrowserAcceleratorKeyEnabled` to `FALSE` this will also change the default value for `AllowBrowserHandle` to `FALSE`. If developers want specific keys to be handled by the browser after changing the `AreBrowserAcceleratorKeyEnabled` settings to `FALSE`, they need to enable them by setting `event_args->put_ AllowBrowserHandle(TRUE)`. This option will give the event arg higher priority over the settings when the browser decide whether to handle the keys.  
+
+# API Details
+## C++
+```
+/// This is a continuation of the ICoreWebView2AcceleratorKeyPressedEventArgs interface.
+[uuid(45238725-3774-4cfe-931f-7985a1b5866f), object, pointer_default(unique)]
+interface ICoreWebView2AcceleratorKeyPressedEventArgs2 : ICoreWebView2AcceleratorKeyPressedEventArgs {
+  /// These APIs allow devs to enable/disable the browser from handling specific accelerator keys.
+  /// By default, `AreBrowserAcceleratorKeyEnabled` is `TRUE` and `AllowBrowserHandle` is `TRUE`.
+  /// When devs set `AreBrowserAcceleratorKeyEnabled` to `FALSE` this will override and the
+  /// default value for `AllowBrowserHandle` will be `FALSE` and prevent the browser to handle
+  /// accelerator keys. If users want specific keys to be handled by the browser after changing
+  /// the settings to `FALSE`, they need to enable them by setting `AllowBrowserHandle` to `TRUE`.
+  /// This option will give the event arg higher priority over the `AreBrowserAcceleratorKeyEnabled`
+  /// settings when we handle the keys.
+  ///
+  /// \snippet ScenarioAcceleratorKeyPressed.cpp
+  /// Gets the `AllowBrowserHandle` property.
+  [propget] HRESULT AllowBrowserHandle([out, retval] BOOL* value);

```

Comment:
The name: From the name I'm not sure what exactly I'm skipping if I disallow it: is it only skipping browser features associated with the accelerator (like how AreBrowserAcceleratorKeyEnabled handles Ctrl+F for find on page, but doesn't cover Ctrl+X for Cut), versus also skipping any feature associated with the accelerator (Ctrl+F _and_ Ctrl+X), versus also skipping the key press showing up in DOM keyboard events (Ctrl+F and Ctrl+X _and also_ the related DOM events aren't raised).

If it matches AreBrowserAcceleratorKeysEnabled we can name it similarly like AllowBrowserAcceleratorKeyHandling.
If its any accelerator key handling then we can name it like AllowDefaultAcceleratorKeyHandling.
And if its everything then we would probably name it Handled but that's already taken. So hopefully this isn't that behavior

Reply:
@bradp0721 We talked about since this matches the subset of accelerator keys that the existing setting does it would be good to match that naming like `AllowBrowserAcceleratorKeyHandling`

Reply:
I completely agree with renaming this to match the existing setting. The setting has Keys in the name as it covers all accelerator keys, so maybe we don't need Key since we're in an event for a specific key.
  AllowBrowserAccelerator
  AllowBrowserAcceleratorHandling
Or we make this the singular version of the setting.
  IsBrowserAcceleratorKeyEnabled
  IsBrowserAcceleratorEnabled